### PR TITLE
fix: prevent DefinitionList from throwing error when null passed in

### PR DESCRIPTION
**Pull request by NewmanJWise**
## 🖼 Context

GPAID-118

This is a small proposal PR to make it so DefinitionList can destructure safely knowing that whatever it is destructing is not null. 

This is useful on pages where you might sometimes optionally not display some definitions. Currently we have to do something like this for optional definitions which just feels not awesome: (it looks a lot worse when there are multiple of them)

```
[
    ...(someObject.reference
      ? [
          {
            title: 'Reference',
            value: someObject.reference,
            key: 'reference',
          },
        ]
      : []),
]
```

## 🚀 Changes

 

Added a tweak filter to ensure that the definitions passed in are 

## 🤔 Considerations

As mentioned above this is just a proposal to make it easier for the consumer to use the component as I would normally except a component using an array to handle a null item in the array without exploding. 

I'm happy not bringing this forward but thought it was worth the discussion :) 

## ✅ Checklist

- [x] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [x] Changes are tested and all tests pass
- [x] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [x] Changes work in all supported browsers (don't forget IE11)
- [ ] You've updated the documentation if necessary

### CODE REVIEW

File: packages/components/src/definitionList/DefinitionList.spec.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ const {
 } = Layout;
 
 describe('DefinitionList', () => {
-  const someDefinitions = () => [{ title: 'First', value: 'first value', key: 'first' }];
+  const someDefinitions = () => [{ title: 'First', value: 'first value', key: 'first' }, null];
```
Comment (NewmanJWise) [inline packages/components/src/definitionList/DefinitionList.spec.js:16]: No need to add additional tests. “hasClass" would blow up in the below tests if a second node was found and previously the component threw an exception when null was passed in but now it works 🎉

**Review by helendurranttw — APPROVED** [review]
🚀

**Review by hendrikmolder — APPROVED** [review]
👏

**Review by oliverviljamaa — APPROVED** [review]
Lovely!

**Pull request merged**